core/src/main/java/cz/xtf/core/config/TestCaseContext.java

```diff
@@ -0,0 +1,47 @@
+package cz.xtf.core.config;
+
+import static cz.xtf.core.config.OpenShiftConfig.OPENSHIFT_NAMESPACE;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class TestCaseContext {
+
+    /**
+     * Used only in case when test cases are executed in different namespaces when:
+     * -Dxtf.openshift.namespace.per.testcase=true
+     *
+     * This allows to track currently running test case for correct namespace mapping. This is used to automatically find
+     * namespace for running test case when
+     * creating {@link cz.xtf.core.openshift.OpenShift} instances.
+     */
+    private static String TEST_CASE_NAME;
+
+    private static final Map<String, String> testcaseToNamespaceMap = new HashMap<String, String>() {
+        {
+            // put default namespace into map to guarantee that it will be deleted with other namespaces
+            put(XTFConfig.get(OPENSHIFT_NAMESPACE), XTFConfig.get(OPENSHIFT_NAMESPACE));
+        }
+    };
+
+    /**
+     * @return test case name associated with current thread or null if not such mapping exists
+     */
+    public static String getTestCaseForCurrentThread() {
```

Comment:
I must admit I do not understand now how `ForCurrentThread` is achieved. 

But couldn't be ThreadLocal use ? https://www.baeldung.com/java-threadlocal

Reply:
Sorry, bad name of the method. I've accidently renamed other one. 

I've decided to step away from `ThreadLocal` as it did fully solved using ForkJoinPool in parallel streams.

Reply:
Can it be set as resolved?